recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,525 @@
+    def _solution_projects(self):
+        solution_path = os.path.join(self._source_subfolder, "PCBuild", "pcbuild.sln")
+        projects = set(m.group(1) for m in re.finditer("\"([^\"]+)\\.vcxproj\"", open(solution_path).read()))
+        discarded = self._msvc_discarded_projects
+
+        def project_build(name):
+            if os.path.basename(name) in discarded:
+                return False
+            if "test" in name:
+                return False
+            return True
+
+        projects = set(p for p in projects if project_build(p))
+        return projects
+
+    @property
+    def _msvc_discarded_projects(self):
+        discarded = {"python_uwp", "pythonw_uwp"}
+        if not self.options.with_bz2:
+            discarded.add("bz2")
+        if not self.options.with_sqlite3:
+            discarded.add("_sqlite3")
+        if not self.options.with_tkinter:
+            discarded.add("_tkinter")
+        if self._is_py2:
+            # Python 2 Visual Studio projects NOT to build
+            discarded = discarded.union({"bdist_wininst", "libeay", "ssleay", "sqlite3", "tcl", "tk", "tix"})
+            if not self.options.with_bsddb:
+                discarded.add("_bsddb")
+        elif self._is_py3:
+            discarded = discarded.union({"bdist_wininst", "liblzma", "openssl", "sqlite3", "xxlimited"})
+            if not self.options.with_lzma:
+                discarded.add("_lzma")
+        return discarded
+
+    @property
+    def _msvc_archs(self):
+        archs = {
+            "x86": "Win32",
+            "x86_64": "x64",
+        }
+        if tools.Version(self.version) >= "3.8":
+            archs.update({
+                "armv7": "ARM",
+                "armv8_32": "ARM",
+                "armv8": "ARM64",
+            })
+        return archs
+
+    def _msvc_build(self):
+        msbuild = MSBuild(self)
+        msbuild_properties = {
+            "IncludeExternals": "true",
+        }
+        projects = self._solution_projects
+        self.output.info("Building {} Visual Studio projects: {}".format(len(projects), projects))
+
+        upgraded = False
+        with tools.no_op():
+            for project_i, project in enumerate(projects):
+                self.output.info("[{}/{}] Building project '{}'...".format(project_i, len(projects), project))
+                project_file = os.path.join(self._source_subfolder, "PCBuild", project + ".vcxproj")
+                msbuild.build(project_file, upgrade_project=not upgraded, build_type="Debug" if self.settings.build_type == "Debug" else "Release",
+                              platforms=self._msvc_archs, properties=msbuild_properties)
+                upgraded = True
+
+    def build(self):
+        if tools.Version(self.version) < "3.9.0":
+            if tools.Version(self.deps_cpp_info["mpdecimal"].version) >= "2.5.0":
+                raise ConanInvalidConfiguration("cpython versions lesser then 3.9.0 require a mpdecimal lesser then 2.5.0")
```

Comment:
I don't think this is really true. This might have been due to a conflict with the vendored mpdecimal that has since been removed. I was at least able to build 3.8.6 against mpdecimal 2.5.0 without issue.

Reply:
Using gcc9@linux, building with mpdecimal 2.5.0 causes compile errors.
The build process as a whole will succeed, but it will end with the following message:
```
Python build finished successfully!
The necessary bits to build these optional modules were not found:
_dbm                  _tkinter              nis                
To find the necessary bits, look in setup.py in detect_modules() for the module's name.


The following modules found by detect_modules() in setup.py, have been
built by the Makefile instead, as configured by the Setup files:
_abc                  atexit                pwd                
time                                                           


Failed to build these modules:
_decimal     
```

I probably should look in a way to treat failed modules as an error.
On the other hand, allowing building modules to fail enables making `mpdecimal` an optional dependency.

<details>

<summary>Copy of _mpdecimal build log</summary>

```
creating build/temp.linux-x86_64-3.7/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal
/usr/bin/gcc -pthread -fPIC -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -m64 -O3 -s -fPIC -m64 -O3 -s -fPIC -std=c99 -Wextra -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -Wno-cast-function-type -Werror=implicit-function-declaration -DCONFIG_64=1 -DASM=1 -Isource_subfolder/Include -IObjects -IPython -I. -I/home/maarten/.conan/data/openssl/1.1.1h/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/home/maarten/.conan/data/expat/2.2.9/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/home/maarten/.conan/data/mpdecimal/2.5.0/_/_/package/ed0afefdcb2cd63a38eaf4030d0a9bc9a0070e9d/include -I/home/maarten/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/home/maarten/.conan/data/libffi/3.3/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/home/maarten/.conan/data/libuuid/1.0.3/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/home/maarten/.conan/data/libuuid/1.0.3/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include/uuid -I/home/maarten/.conan/data/libxcrypt/4.4.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/home/maarten/.conan/data/bzip2/1.0.8/_/_/package/da606cf731e334010b0bf6e85a2a6f891b9f36b0/include -I/home/maarten/.conan/data/gdbm/1.18.1/_/_/package/c5fadfd295e8ca62e60af52df5045a8a9dc60339/include -I/home/maarten/.conan/data/sqlite3/3.33.0/_/_/package/01a8a96da981e1edfc87074a67d005b4b5f9950d/include -I/home/maarten/.conan/data/ncurses/6.2/_/_/package/66d8b941c19564a5b3f22eeb061a3d72ffc41c23/include -I/home/maarten/.conan/data/ncurses/6.2/_/_/package/66d8b941c19564a5b3f22eeb061a3d72ffc41c23/include/ncursesw -I/home/maarten/.conan/data/xz_utils/5.2.5/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include -I/usr/local/include -I/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Include -I/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6 -c /home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c -o build/temp.linux-x86_64-3.7/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.o
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:279:36: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  279 | dec_traphandler(mpd_context_t *ctx UNUSED) /* GCOV_NOT_REACHED */
      |                                    ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:539:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  539 | signaldict_init(PyObject *self, PyObject *args UNUSED, PyObject *kwds UNUSED)
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:546:31: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  546 | signaldict_len(PyObject *self UNUSED)
      |                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:553:32: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  553 | signaldict_iter(PyObject *self UNUSED)
      |                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:657:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  657 | signaldict_copy(PyObject *self, PyObject *args UNUSED)
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:664:14: error: ‘signaldict_len’ undeclared here (not in a function); did you mean ‘signaldict_repr’?
  664 |     (lenfunc)signaldict_len,          /* mp_length */
      |              ^~~~~~~~~~~~~~
      |              signaldict_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:670:28: error: ‘signaldict_copy’ undeclared here (not in a function); did you mean ‘signaldict_repr’?
  670 |     { "copy", (PyCFunction)signaldict_copy, METH_NOARGS, NULL},
      |                            ^~~~~~~~~~~~~~~
      |                            signaldict_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:703:18: error: ‘signaldict_iter’ undeclared here (not in a function); did you mean ‘signaldict_repr’?
  703 |     (getiterfunc)signaldict_iter,             /* tp_iter */
      |                  ^~~~~~~~~~~~~~~
      |                  signaldict_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:703:5: error: initializer element is not constant
  703 |     (getiterfunc)signaldict_iter,             /* tp_iter */
      |     ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:703:5: note: (near initialization for ‘PyDecSignalDictMixin_Type.tp_iter’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:713:15: error: ‘signaldict_init’ undeclared here (not in a function); did you mean ‘signaldict_repr’?
  713 |     (initproc)signaldict_init,                /* tp_init */
      |               ^~~~~~~~~~~~~~~
      |               signaldict_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:713:5: error: initializer element is not constant
  713 |     (initproc)signaldict_init,                /* tp_init */
      |     ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:713:5: note: (near initialization for ‘PyDecSignalDictMixin_Type.tp_init’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:725:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  725 | context_get##mem(PyObject *self, void *closure UNUSED)  \
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:737:1: note: in expansion of macro ‘Dec_CONTEXT_GET_SSIZE’
  737 | Dec_CONTEXT_GET_SSIZE(prec)
      | ^~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:725:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  725 | context_get##mem(PyObject *self, void *closure UNUSED)  \
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:738:1: note: in expansion of macro ‘Dec_CONTEXT_GET_SSIZE’
  738 | Dec_CONTEXT_GET_SSIZE(emax)
      | ^~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:725:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  725 | context_get##mem(PyObject *self, void *closure UNUSED)  \
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:739:1: note: in expansion of macro ‘Dec_CONTEXT_GET_SSIZE’
  739 | Dec_CONTEXT_GET_SSIZE(emin)
      | ^~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:725:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  725 | context_get##mem(PyObject *self, void *closure UNUSED)  \
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:740:1: note: in expansion of macro ‘Dec_CONTEXT_GET_SSIZE’
  740 | Dec_CONTEXT_GET_SSIZE(clamp)
      | ^~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:748:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  748 | context_getround(PyObject *self, void *closure UNUSED)
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:757:51: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  757 | context_getcapitals(PyObject *self, void *closure UNUSED)
      |                                                   ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:771:50: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  771 | context_getetiny(PyObject *self, PyObject *dummy UNUSED)
      |                                                  ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:777:49: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  777 | context_getetop(PyObject *self, PyObject *dummy UNUSED)
      |                                                 ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:783:64: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  783 | context_setprec(PyObject *self, PyObject *value, void *closure UNUSED)
      |                                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:803:64: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  803 | context_setemin(PyObject *self, PyObject *value, void *closure UNUSED)
      |                                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:823:64: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  823 | context_setemax(PyObject *self, PyObject *value, void *closure UNUSED)
      |                                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:905:65: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  905 | context_setround(PyObject *self, PyObject *value, void *closure UNUSED)
      |                                                                 ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:924:68: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
  924 | context_setcapitals(PyObject *self, PyObject *value, void *closure UNUSED)
      |                                                                    ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1071:65: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1071 | context_setclamp(PyObject *self, PyObject *value, void *closure UNUSED)
      |                                                                 ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1155:53: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1155 | context_clear_traps(PyObject *self, PyObject *dummy UNUSED)
      |                                                     ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1162:53: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1162 | context_clear_flags(PyObject *self, PyObject *dummy UNUSED)
      |                                                     ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1178:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1178 | context_new(PyTypeObject *type, PyObject *args UNUSED, PyObject *kwds UNUSED)
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: In function ‘context_init’:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1265:28: error: implicit declaration of function ‘context_setprec’; did you mean ‘context_setattr’? [-Werror=implicit-function-declaration]
 1265 |     if (prec != Py_None && context_setprec(self, prec, NULL) < 0) {
      |                            ^~~~~~~~~~~~~~~
      |                            context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1268:32: error: implicit declaration of function ‘context_setround’; did you mean ‘context_setattr’? [-Werror=implicit-function-declaration]
 1268 |     if (rounding != Py_None && context_setround(self, rounding, NULL) < 0) {
      |                                ^~~~~~~~~~~~~~~~
      |                                context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1271:28: error: implicit declaration of function ‘context_setemin’; did you mean ‘context_setattr’? [-Werror=implicit-function-declaration]
 1271 |     if (emin != Py_None && context_setemin(self, emin, NULL) < 0) {
      |                            ^~~~~~~~~~~~~~~
      |                            context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1274:28: error: implicit declaration of function ‘context_setemax’; did you mean ‘context_setattr’? [-Werror=implicit-function-declaration]
 1274 |     if (emax != Py_None && context_setemax(self, emax, NULL) < 0) {
      |                            ^~~~~~~~~~~~~~~
      |                            context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1277:32: error: implicit declaration of function ‘context_setcapitals’; did you mean ‘context_setattr’? [-Werror=implicit-function-declaration]
 1277 |     if (capitals != Py_None && context_setcapitals(self, capitals, NULL) < 0) {
      |                                ^~~~~~~~~~~~~~~~~~~
      |                                context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1280:29: error: implicit declaration of function ‘context_setclamp’; did you mean ‘context_setattr’? [-Werror=implicit-function-declaration]
 1280 |     if (clamp != Py_None && context_setclamp(self, clamp, NULL) < 0) {
      |                             ^~~~~~~~~~~~~~~~
      |                             context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: At top level:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1412:45: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1412 | context_copy(PyObject *self, PyObject *args UNUSED)
      |                                             ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1429:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1429 | context_reduce(PyObject *self, PyObject *args UNUSED)
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1463:21: error: ‘context_getprec’ undeclared here (not in a function); did you mean ‘context_getsets’?
 1463 |   { "prec", (getter)context_getprec, (setter)context_setprec, NULL, NULL},
      |                     ^~~~~~~~~~~~~~~
      |                     context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1463:13: error: initializer element is not constant
 1463 |   { "prec", (getter)context_getprec, (setter)context_setprec, NULL, NULL},
      |             ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1463:13: note: (near initialization for ‘context_getsets[0].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1463:46: error: ‘context_setprec’ undeclared here (not in a function); did you mean ‘context_repr’?
 1463 |   { "prec", (getter)context_getprec, (setter)context_setprec, NULL, NULL},
      |                                              ^~~~~~~~~~~~~~~
      |                                              context_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1463:38: error: initializer element is not constant
 1463 |   { "prec", (getter)context_getprec, (setter)context_setprec, NULL, NULL},
      |                                      ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1463:38: note: (near initialization for ‘context_getsets[0].set’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1464:21: error: ‘context_getemax’ undeclared here (not in a function); did you mean ‘context_getsets’?
 1464 |   { "Emax", (getter)context_getemax, (setter)context_setemax, NULL, NULL},
      |                     ^~~~~~~~~~~~~~~
      |                     context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1464:13: error: initializer element is not constant
 1464 |   { "Emax", (getter)context_getemax, (setter)context_setemax, NULL, NULL},
      |             ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1464:13: note: (near initialization for ‘context_getsets[1].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1464:46: error: ‘context_setemax’ undeclared here (not in a function); did you mean ‘context_setattr’?
 1464 |   { "Emax", (getter)context_getemax, (setter)context_setemax, NULL, NULL},
      |                                              ^~~~~~~~~~~~~~~
      |                                              context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1464:38: error: initializer element is not constant
 1464 |   { "Emax", (getter)context_getemax, (setter)context_setemax, NULL, NULL},
      |                                      ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1464:38: note: (near initialization for ‘context_getsets[1].set’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1465:21: error: ‘context_getemin’ undeclared here (not in a function); did you mean ‘context_getsets’?
 1465 |   { "Emin", (getter)context_getemin, (setter)context_setemin, NULL, NULL},
      |                     ^~~~~~~~~~~~~~~
      |                     context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1465:13: error: initializer element is not constant
 1465 |   { "Emin", (getter)context_getemin, (setter)context_setemin, NULL, NULL},
      |             ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1465:13: note: (near initialization for ‘context_getsets[2].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1465:46: error: ‘context_setemin’ undeclared here (not in a function); did you mean ‘context_setattr’?
 1465 |   { "Emin", (getter)context_getemin, (setter)context_setemin, NULL, NULL},
      |                                              ^~~~~~~~~~~~~~~
      |                                              context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1465:38: error: initializer element is not constant
 1465 |   { "Emin", (getter)context_getemin, (setter)context_setemin, NULL, NULL},
      |                                      ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1465:38: note: (near initialization for ‘context_getsets[2].set’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1466:25: error: ‘context_getround’ undeclared here (not in a function); did you mean ‘context_getsets’?
 1466 |   { "rounding", (getter)context_getround, (setter)context_setround, NULL, NULL},
      |                         ^~~~~~~~~~~~~~~~
      |                         context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1466:17: error: initializer element is not constant
 1466 |   { "rounding", (getter)context_getround, (setter)context_setround, NULL, NULL},
      |                 ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1466:17: note: (near initialization for ‘context_getsets[3].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1466:51: error: ‘context_setround’ undeclared here (not in a function); did you mean ‘context_setattr’?
 1466 |   { "rounding", (getter)context_getround, (setter)context_setround, NULL, NULL},
      |                                                   ^~~~~~~~~~~~~~~~
      |                                                   context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1466:43: error: initializer element is not constant
 1466 |   { "rounding", (getter)context_getround, (setter)context_setround, NULL, NULL},
      |                                           ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1466:43: note: (near initialization for ‘context_getsets[3].set’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1467:25: error: ‘context_getcapitals’ undeclared here (not in a function); did you mean ‘context_getsets’?
 1467 |   { "capitals", (getter)context_getcapitals, (setter)context_setcapitals, NULL, NULL},
      |                         ^~~~~~~~~~~~~~~~~~~
      |                         context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1467:17: error: initializer element is not constant
 1467 |   { "capitals", (getter)context_getcapitals, (setter)context_setcapitals, NULL, NULL},
      |                 ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1467:17: note: (near initialization for ‘context_getsets[4].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1467:54: error: ‘context_setcapitals’ undeclared here (not in a function); did you mean ‘context_setattr’?
 1467 |   { "capitals", (getter)context_getcapitals, (setter)context_setcapitals, NULL, NULL},
      |                                                      ^~~~~~~~~~~~~~~~~~~
      |                                                      context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1467:46: error: initializer element is not constant
 1467 |   { "capitals", (getter)context_getcapitals, (setter)context_setcapitals, NULL, NULL},
      |                                              ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1467:46: note: (near initialization for ‘context_getsets[4].set’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1468:22: error: ‘context_getclamp’ undeclared here (not in a function); did you mean ‘context_getsets’?
 1468 |   { "clamp", (getter)context_getclamp, (setter)context_setclamp, NULL, NULL},
      |                      ^~~~~~~~~~~~~~~~
      |                      context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1468:14: error: initializer element is not constant
 1468 |   { "clamp", (getter)context_getclamp, (setter)context_setclamp, NULL, NULL},
      |              ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1468:14: note: (near initialization for ‘context_getsets[5].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1468:48: error: ‘context_setclamp’ undeclared here (not in a function); did you mean ‘context_setattr’?
 1468 |   { "clamp", (getter)context_getclamp, (setter)context_setclamp, NULL, NULL},
      |                                                ^~~~~~~~~~~~~~~~
      |                                                context_setattr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1468:40: error: initializer element is not constant
 1468 |   { "clamp", (getter)context_getclamp, (setter)context_setclamp, NULL, NULL},
      |                                        ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1468:40: note: (near initialization for ‘context_getsets[5].set’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: In function ‘init_current_context’:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1638:28: error: implicit declaration of function ‘context_copy’; did you mean ‘context_repr’? [-Werror=implicit-function-declaration]
 1638 |     PyObject *tl_context = context_copy(default_context_template, NULL);
      |                            ^~~~~~~~~~~~
      |                            context_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1638:28: warning: initialization of ‘PyObject *’ {aka ‘struct _object *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: At top level:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1679:40: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1679 | PyDec_GetCurrentContext(PyObject *self UNUSED, PyObject *args UNUSED)
      |                                        ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1686:40: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1686 | PyDec_SetCurrentContext(PyObject *self UNUSED, PyObject *v)
      |                                        ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1720:35: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1720 | ctxmanager_new(PyTypeObject *type UNUSED, PyObject *args, PyObject *kwds)
      |                                   ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1767:70: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1767 | ctxmanager_set_local(PyDecContextManagerObject *self, PyObject *args UNUSED)
      |                                                                      ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1783:42: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 1783 |                           PyObject *args UNUSED)
      |                                          ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1798:30: error: ‘ctxmanager_set_local’ undeclared here (not in a function); did you mean ‘ctxmanager_dealloc’?
 1798 |   {"__enter__", (PyCFunction)ctxmanager_set_local, METH_NOARGS, NULL},
      |                              ^~~~~~~~~~~~~~~~~~~~
      |                              ctxmanager_dealloc
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1798:17: error: initializer element is not constant
 1798 |   {"__enter__", (PyCFunction)ctxmanager_set_local, METH_NOARGS, NULL},
      |                 ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1798:17: note: (near initialization for ‘ctxmanager_methods[0].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1799:29: error: ‘ctxmanager_restore_global’ undeclared here (not in a function)
 1799 |   {"__exit__", (PyCFunction)ctxmanager_restore_global, METH_VARARGS, NULL},
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1799:16: error: initializer element is not constant
 1799 |   {"__exit__", (PyCFunction)ctxmanager_restore_global, METH_VARARGS, NULL},
      |                ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:1799:16: note: (near initialization for ‘ctxmanager_methods[1].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3409:53: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3409 | dec_as_integer_ratio(PyObject *self, PyObject *args UNUSED)
      |                                                     ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3671:46: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3671 | PyDec_AsTuple(PyObject *dec, PyObject *dummy UNUSED)
      |                                              ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4128:1: note: in expansion of macro ‘Dec_BoolFunc’
 4128 | Dec_BoolFunc(mpd_iscanonical)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4129:1: note: in expansion of macro ‘Dec_BoolFunc’
 4129 | Dec_BoolFunc(mpd_isfinite)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4130:1: note: in expansion of macro ‘Dec_BoolFunc’
 4130 | Dec_BoolFunc(mpd_isinfinite)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4131:1: note: in expansion of macro ‘Dec_BoolFunc’
 4131 | Dec_BoolFunc(mpd_isnan)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4132:1: note: in expansion of macro ‘Dec_BoolFunc’
 4132 | Dec_BoolFunc(mpd_isqnan)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4133:1: note: in expansion of macro ‘Dec_BoolFunc’
 4133 | Dec_BoolFunc(mpd_issnan)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4134:1: note: in expansion of macro ‘Dec_BoolFunc’
 4134 | Dec_BoolFunc(mpd_issigned)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:3824:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 3824 | dec_##MPDFUNC(PyObject *self, PyObject *dummy UNUSED)       \
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4135:1: note: in expansion of macro ‘Dec_BoolFunc’
 4135 | Dec_BoolFunc(mpd_iszero)
      | ^~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4143:48: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4143 | dec_mpd_adjexp(PyObject *self, PyObject *dummy UNUSED)
      |                                                ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4158:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4158 | dec_canonical(PyObject *self, PyObject *dummy UNUSED)
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4165:47: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4165 | dec_conjugate(PyObject *self, PyObject *dummy UNUSED)
      |                                               ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4172:30: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4172 | dec_mpd_radix(PyObject *self UNUSED, PyObject *dummy UNUSED)
      |                              ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4186:51: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4186 | dec_mpd_qcopy_abs(PyObject *self, PyObject *dummy UNUSED)
      |                                                   ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4206:54: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4206 | dec_mpd_qcopy_negate(PyObject *self, PyObject *dummy UNUSED)
      |                                                      ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4452:42: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4452 | dec_ceil(PyObject *self, PyObject *dummy UNUSED)
      |                                          ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4462:45: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4462 | dec_complex(PyObject *self, PyObject *dummy UNUSED)
      |                                             ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4483:42: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4483 | dec_copy(PyObject *self, PyObject *dummy UNUSED)
      |                                          ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4491:43: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4491 | dec_floor(PyObject *self, PyObject *dummy UNUSED)
      |                                           ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4628:44: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4628 | dec_reduce(PyObject *self, PyObject *dummy UNUSED)
      |                                            ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4645:41: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4645 | dec_sizeof(PyObject *v, PyObject *dummy UNUSED)
      |                                         ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4658:43: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4658 | dec_trunc(PyObject *self, PyObject *dummy UNUSED)
      |                                           ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4668:40: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4668 | dec_real(PyObject *self, void *closure UNUSED)
      |                                        ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4675:25: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 4675 | dec_imag(PyObject *self UNUSED, void *closure UNUSED)
      |                         ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4691:21: error: ‘dec_real’ undeclared here (not in a function); did you mean ‘dec_repr’?
 4691 |   { "real", (getter)dec_real, NULL, NULL, NULL},
      |                     ^~~~~~~~
      |                     dec_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4691:13: error: initializer element is not constant
 4691 |   { "real", (getter)dec_real, NULL, NULL, NULL},
      |             ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4691:13: note: (near initialization for ‘dec_getsets[0].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4692:21: error: ‘dec_imag’ undeclared here (not in a function)
 4692 |   { "imag", (getter)dec_imag, NULL, NULL, NULL},
      |                     ^~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4692:13: error: initializer element is not constant
 4692 |   { "imag", (getter)dec_imag, NULL, NULL, NULL},
      |             ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4692:13: note: (near initialization for ‘dec_getsets[1].get’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4762:21: error: ‘dec_mpd_iscanonical’ undeclared here (not in a function); did you mean ‘mpd_iscanonical’?
 4762 |   { "is_canonical", dec_mpd_iscanonical, METH_NOARGS, doc_is_canonical },
      |                     ^~~~~~~~~~~~~~~~~~~
      |                     mpd_iscanonical
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4762:21: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4762:21: note: (near initialization for ‘dec_methods[20].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4762:21: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4762:21: note: (near initialization for ‘dec_methods[20].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4763:18: error: ‘dec_mpd_isfinite’ undeclared here (not in a function); did you mean ‘mpd_isfinite’?
 4763 |   { "is_finite", dec_mpd_isfinite, METH_NOARGS, doc_is_finite },
      |                  ^~~~~~~~~~~~~~~~
      |                  mpd_isfinite
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4763:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4763:18: note: (near initialization for ‘dec_methods[21].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4763:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4763:18: note: (near initialization for ‘dec_methods[21].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4764:20: error: ‘dec_mpd_isinfinite’ undeclared here (not in a function); did you mean ‘mpd_isinfinite’?
 4764 |   { "is_infinite", dec_mpd_isinfinite, METH_NOARGS, doc_is_infinite },
      |                    ^~~~~~~~~~~~~~~~~~
      |                    mpd_isinfinite
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4764:20: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4764:20: note: (near initialization for ‘dec_methods[22].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4764:20: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4764:20: note: (near initialization for ‘dec_methods[22].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4765:15: error: ‘dec_mpd_isnan’ undeclared here (not in a function); did you mean ‘dec_mpd_qand’?
 4765 |   { "is_nan", dec_mpd_isnan, METH_NOARGS, doc_is_nan },
      |               ^~~~~~~~~~~~~
      |               dec_mpd_qand
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4765:15: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4765:15: note: (near initialization for ‘dec_methods[23].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4765:15: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4765:15: note: (near initialization for ‘dec_methods[23].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4766:16: error: ‘dec_mpd_isqnan’ undeclared here (not in a function); did you mean ‘dec_mpd_qand’?
 4766 |   { "is_qnan", dec_mpd_isqnan, METH_NOARGS, doc_is_qnan },
      |                ^~~~~~~~~~~~~~
      |                dec_mpd_qand
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4766:16: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4766:16: note: (near initialization for ‘dec_methods[24].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4766:16: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4766:16: note: (near initialization for ‘dec_methods[24].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4767:16: error: ‘dec_mpd_issnan’ undeclared here (not in a function); did you mean ‘mpd_issnan’?
 4767 |   { "is_snan", dec_mpd_issnan, METH_NOARGS, doc_is_snan },
      |                ^~~~~~~~~~~~~~
      |                mpd_issnan
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4767:16: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4767:16: note: (near initialization for ‘dec_methods[25].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4767:16: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4767:16: note: (near initialization for ‘dec_methods[25].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4768:18: error: ‘dec_mpd_issigned’ undeclared here (not in a function); did you mean ‘mpd_issigned’?
 4768 |   { "is_signed", dec_mpd_issigned, METH_NOARGS, doc_is_signed },
      |                  ^~~~~~~~~~~~~~~~
      |                  mpd_issigned
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4768:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4768:18: note: (near initialization for ‘dec_methods[26].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4768:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4768:18: note: (near initialization for ‘dec_methods[26].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4769:16: error: ‘dec_mpd_iszero’ undeclared here (not in a function); did you mean ‘dec_mpd_qinvert’?
 4769 |   { "is_zero", dec_mpd_iszero, METH_NOARGS, doc_is_zero },
      |                ^~~~~~~~~~~~~~
      |                dec_mpd_qinvert
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4769:16: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4769:16: note: (near initialization for ‘dec_methods[27].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4769:16: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4769:16: note: (near initialization for ‘dec_methods[27].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4776:17: error: ‘dec_mpd_adjexp’ undeclared here (not in a function); did you mean ‘dec_mpd_qexp’?
 4776 |   { "adjusted", dec_mpd_adjexp, METH_NOARGS, doc_adjusted },
      |                 ^~~~~~~~~~~~~~
      |                 dec_mpd_qexp
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4776:17: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4776:17: note: (near initialization for ‘dec_methods[30].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4776:17: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4776:17: note: (near initialization for ‘dec_methods[30].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4777:18: error: ‘dec_canonical’ undeclared here (not in a function); did you mean ‘doc_canonical’?
 4777 |   { "canonical", dec_canonical, METH_NOARGS, doc_canonical },
      |                  ^~~~~~~~~~~~~
      |                  doc_canonical
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4777:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4777:18: note: (near initialization for ‘dec_methods[31].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4777:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4777:18: note: (near initialization for ‘dec_methods[31].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4778:18: error: ‘dec_conjugate’ undeclared here (not in a function); did you mean ‘doc_conjugate’?
 4778 |   { "conjugate", dec_conjugate, METH_NOARGS, doc_conjugate },
      |                  ^~~~~~~~~~~~~
      |                  doc_conjugate
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4778:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4778:18: note: (near initialization for ‘dec_methods[32].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4778:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4778:18: note: (near initialization for ‘dec_methods[32].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4779:14: error: ‘dec_mpd_radix’ undeclared here (not in a function); did you mean ‘dec_mpd_qand’?
 4779 |   { "radix", dec_mpd_radix, METH_NOARGS, doc_radix },
      |              ^~~~~~~~~~~~~
      |              dec_mpd_qand
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4779:14: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4779:14: note: (near initialization for ‘dec_methods[33].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4779:14: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4779:14: note: (near initialization for ‘dec_methods[33].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4782:17: error: ‘dec_mpd_qcopy_abs’ undeclared here (not in a function); did you mean ‘dec_mpd_qcopy_sign’?
 4782 |   { "copy_abs", dec_mpd_qcopy_abs, METH_NOARGS, doc_copy_abs },
      |                 ^~~~~~~~~~~~~~~~~
      |                 dec_mpd_qcopy_sign
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4782:17: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4782:17: note: (near initialization for ‘dec_methods[34].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4782:17: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4782:17: note: (near initialization for ‘dec_methods[34].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4783:20: error: ‘dec_mpd_qcopy_negate’ undeclared here (not in a function); did you mean ‘mpd_qcopy_negate’?
 4783 |   { "copy_negate", dec_mpd_qcopy_negate, METH_NOARGS, doc_copy_negate },
      |                    ^~~~~~~~~~~~~~~~~~~~
      |                    mpd_qcopy_negate
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4783:20: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4783:20: note: (near initialization for ‘dec_methods[35].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4783:20: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4783:20: note: (near initialization for ‘dec_methods[35].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4807:17: error: ‘PyDec_AsTuple’ undeclared here (not in a function); did you mean ‘PyDec_Type’?
 4807 |   { "as_tuple", PyDec_AsTuple, METH_NOARGS, doc_as_tuple },
      |                 ^~~~~~~~~~~~~
      |                 PyDec_Type
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4807:17: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4807:17: note: (near initialization for ‘dec_methods[51].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4807:17: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4807:17: note: (near initialization for ‘dec_methods[51].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4808:25: error: ‘dec_as_integer_ratio’ undeclared here (not in a function); did you mean ‘doc_as_integer_ratio’?
 4808 |   { "as_integer_ratio", dec_as_integer_ratio, METH_NOARGS, doc_as_integer_ratio },
      |                         ^~~~~~~~~~~~~~~~~~~~
      |                         doc_as_integer_ratio
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4808:25: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4808:25: note: (near initialization for ‘dec_methods[52].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4808:25: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4808:25: note: (near initialization for ‘dec_methods[52].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4811:17: error: ‘dec_copy’ undeclared here (not in a function)
 4811 |   { "__copy__", dec_copy, METH_NOARGS, NULL },
      |                 ^~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4811:17: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4811:17: note: (near initialization for ‘dec_methods[53].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4811:17: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4811:17: note: (near initialization for ‘dec_methods[53].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4812:21: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
 4812 |   { "__deepcopy__", dec_copy, METH_O, NULL },
      |                     ^~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4812:21: note: (near initialization for ‘dec_methods[54].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4812:21: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4812:21: note: (near initialization for ‘dec_methods[54].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4814:19: error: ‘dec_reduce’ undeclared here (not in a function); did you mean ‘mpd_reduce’?
 4814 |   { "__reduce__", dec_reduce, METH_NOARGS, NULL },
      |                   ^~~~~~~~~~
      |                   mpd_reduce
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4814:19: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4814:19: note: (near initialization for ‘dec_methods[56].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4814:19: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4814:19: note: (near initialization for ‘dec_methods[56].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4816:17: error: ‘dec_ceil’ undeclared here (not in a function)
 4816 |   { "__ceil__", dec_ceil, METH_NOARGS, NULL },
      |                 ^~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4816:17: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4816:17: note: (near initialization for ‘dec_methods[58].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4816:17: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4816:17: note: (near initialization for ‘dec_methods[58].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4817:18: error: ‘dec_floor’ undeclared here (not in a function); did you mean ‘mpd_floor’?
 4817 |   { "__floor__", dec_floor, METH_NOARGS, NULL },
      |                  ^~~~~~~~~
      |                  mpd_floor
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4817:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4817:18: note: (near initialization for ‘dec_methods[59].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4817:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4817:18: note: (near initialization for ‘dec_methods[59].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4818:18: error: ‘dec_trunc’ undeclared here (not in a function); did you mean ‘mpd_trunc’?
 4818 |   { "__trunc__", dec_trunc, METH_NOARGS, NULL },
      |                  ^~~~~~~~~
      |                  mpd_trunc
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4818:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4818:18: note: (near initialization for ‘dec_methods[60].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4818:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4818:18: note: (near initialization for ‘dec_methods[60].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4819:20: error: ‘dec_complex’ undeclared here (not in a function); did you mean ‘Py_complex’?
 4819 |   { "__complex__", dec_complex, METH_NOARGS, NULL },
      |                    ^~~~~~~~~~~
      |                    Py_complex
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4819:20: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4819:20: note: (near initialization for ‘dec_methods[61].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4819:20: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4819:20: note: (near initialization for ‘dec_methods[61].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4820:19: error: ‘dec_sizeof’ undeclared here (not in a function)
 4820 |   { "__sizeof__", dec_sizeof, METH_NOARGS, NULL },
      |                   ^~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4820:19: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyGetSetDef *’ {aka ‘struct PyGetSetDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4820:19: note: (near initialization for ‘dec_methods[62].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4820:19: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:4820:19: note: (near initialization for ‘dec_methods[62].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: In function ‘ctx_mpd_radix’:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5168:12: error: implicit declaration of function ‘dec_mpd_radix’; did you mean ‘ctx_mpd_radix’? [-Werror=implicit-function-declaration]
 5168 |     return dec_mpd_radix(context, dummy);
      |            ^~~~~~~~~~~~~
      |            ctx_mpd_radix
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5168:12: warning: returning ‘int’ from a function with return type ‘PyObject *’ {aka ‘struct _object *’} makes pointer from integer without a cast [-Wint-conversion]
 5168 |     return dec_mpd_radix(context, dummy);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: At top level:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5183:35: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 5183 | ctx_iscanonical(PyObject *context UNUSED, PyObject *v)
      |                                   ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5208:33: error: expected ‘;’, ‘,’ or ‘)’ before ‘UNUSED’
 5208 | ctx_canonical(PyObject *context UNUSED, PyObject *v)
      |                                 ^~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5447:14: error: ‘context_getetiny’ undeclared here (not in a function); did you mean ‘context_getsets’?
 5447 |   { "Etiny", context_getetiny, METH_NOARGS, doc_ctx_Etiny },
      |              ^~~~~~~~~~~~~~~~
      |              context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5447:14: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5447:14: note: (near initialization for ‘context_methods[31].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5447:14: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5447:14: note: (near initialization for ‘context_methods[31].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5448:13: error: ‘context_getetop’ undeclared here (not in a function); did you mean ‘context_getsets’?
 5448 |   { "Etop", context_getetop, METH_NOARGS, doc_ctx_Etop },
      |             ^~~~~~~~~~~~~~~
      |             context_getsets
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5448:13: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5448:13: note: (near initialization for ‘context_methods[32].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5448:13: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5448:13: note: (near initialization for ‘context_methods[32].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5452:21: error: ‘ctx_iscanonical’ undeclared here (not in a function); did you mean ‘mpd_iscanonical’?
 5452 |   { "is_canonical", ctx_iscanonical, METH_O, doc_ctx_is_canonical },
      |                     ^~~~~~~~~~~~~~~
      |                     mpd_iscanonical
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5452:21: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5452:21: note: (near initialization for ‘context_methods[34].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5452:21: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5452:21: note: (near initialization for ‘context_methods[34].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5468:18: error: ‘ctx_canonical’ undeclared here (not in a function); did you mean ‘doc_canonical’?
 5468 |   { "canonical", ctx_canonical, METH_O, doc_ctx_canonical },
      |                  ^~~~~~~~~~~~~
      |                  doc_canonical
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5468:18: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5468:18: note: (near initialization for ‘context_methods[45].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5468:18: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5468:18: note: (near initialization for ‘context_methods[45].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5491:20: error: ‘context_clear_flags’ undeclared here (not in a function); did you mean ‘doc_ctx_clear_flags’?
 5491 |   { "clear_flags", context_clear_flags, METH_NOARGS, doc_ctx_clear_flags },
      |                    ^~~~~~~~~~~~~~~~~~~
      |                    doc_ctx_clear_flags
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5491:20: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5491:20: note: (near initialization for ‘context_methods[64].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5491:20: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5491:20: note: (near initialization for ‘context_methods[64].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5492:20: error: ‘context_clear_traps’ undeclared here (not in a function); did you mean ‘doc_ctx_clear_traps’?
 5492 |   { "clear_traps", context_clear_traps, METH_NOARGS, doc_ctx_clear_traps },
      |                    ^~~~~~~~~~~~~~~~~~~
      |                    doc_ctx_clear_traps
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5492:20: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5492:20: note: (near initialization for ‘context_methods[65].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5492:20: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5492:20: note: (near initialization for ‘context_methods[65].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5502:30: error: ‘context_copy’ undeclared here (not in a function); did you mean ‘context_repr’?
 5502 |   { "__copy__", (PyCFunction)context_copy, METH_NOARGS, NULL },
      |                              ^~~~~~~~~~~~
      |                              context_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5502:17: error: initializer element is not constant
 5502 |   { "__copy__", (PyCFunction)context_copy, METH_NOARGS, NULL },
      |                 ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5502:17: note: (near initialization for ‘context_methods[66].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5503:19: error: ‘context_reduce’ undeclared here (not in a function); did you mean ‘context_repr’?
 5503 |   { "__reduce__", context_reduce, METH_NOARGS, NULL },
      |                   ^~~~~~~~~~~~~~
      |                   context_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5503:19: warning: initialization of ‘PyObject * (*)(PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5503:19: note: (near initialization for ‘context_methods[67].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5503:19: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5503:19: note: (near initialization for ‘context_methods[67].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5504:13: error: initializer element is not constant
 5504 |   { "copy", (PyCFunction)context_copy, METH_NOARGS, doc_ctx_copy },
      |             ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5504:13: note: (near initialization for ‘context_methods[68].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5550:5: error: ‘context_new’ undeclared here (not in a function); did you mean ‘context_repr’?
 5550 |     context_new,                               /* tp_new */
      |     ^~~~~~~~~~~
      |     context_repr
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5550:5: warning: initialization of ‘PyObject * (*)(struct _typeobject *, PyObject *, PyObject *)’ {aka ‘struct _object * (*)(struct _typeobject *, struct _object *, struct _object *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5550:5: note: (near initialization for ‘PyDecContext_Type.tp_new’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5550:5: error: initializer element is not constant
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5550:5: note: (near initialization for ‘PyDecContext_Type.tp_new’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5557:32: error: ‘PyDec_GetCurrentContext’ undeclared here (not in a function)
 5557 |   { "getcontext", (PyCFunction)PyDec_GetCurrentContext, METH_NOARGS, doc_getcontext},
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5557:19: error: initializer element is not constant
 5557 |   { "getcontext", (PyCFunction)PyDec_GetCurrentContext, METH_NOARGS, doc_getcontext},
      |                   ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5557:19: note: (near initialization for ‘_decimal_methods[0].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5558:32: error: ‘PyDec_SetCurrentContext’ undeclared here (not in a function)
 5558 |   { "setcontext", (PyCFunction)PyDec_SetCurrentContext, METH_O, doc_setcontext},
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5558:19: error: initializer element is not constant
 5558 |   { "setcontext", (PyCFunction)PyDec_SetCurrentContext, METH_O, doc_setcontext},
      |                   ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5558:19: note: (near initialization for ‘_decimal_methods[1].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5559:34: error: ‘ctxmanager_new’ undeclared here (not in a function); did you mean ‘ctxmanager_methods’?
 5559 |   { "localcontext", (PyCFunction)ctxmanager_new, METH_VARARGS|METH_KEYWORDS, doc_localcontext},
      |                                  ^~~~~~~~~~~~~~
      |                                  ctxmanager_methods
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5559:21: error: initializer element is not constant
 5559 |   { "localcontext", (PyCFunction)ctxmanager_new, METH_VARARGS|METH_KEYWORDS, doc_localcontext},
      |                     ^
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5559:21: note: (near initialization for ‘_decimal_methods[2].ml_meth’)
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c: In function ‘PyInit__decimal’:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5669:23: error: ‘dec_traphandler’ undeclared (first use in this function); did you mean ‘mpd_traphandler’?
 5669 |     mpd_traphandler = dec_traphandler;
      |                       ^~~~~~~~~~~~~~~
      |                       mpd_traphandler
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5669:23: note: each undeclared identifier is reported only once for each function it appears in
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:5669:21: warning: assignment to ‘void (*)(mpd_context_t *)’ {aka ‘void (*)(struct mpd_context_t *)’} from incompatible pointer type ‘PyMethodDef *’ {aka ‘struct PyMethodDef *’} [-Wincompatible-pointer-types]
 5669 |     mpd_traphandler = dec_traphandler;
      |                     ^
At top level:
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:393:1: warning: ‘flags_as_dict’ defined but not used [-Wunused-function]
  393 | flags_as_dict(uint32_t flags)
      | ^~~~~~~~~~~~~
/home/maarten/.conan/data/cpython/3.7.9/_/_/build/8903d058fc58317b994f892434fb1aa9372734f6/source_subfolder/Modules/_decimal/_decimal.c:347:1: warning: ‘signals_as_list’ defined but not used [-Wunused-function]
  347 | signals_as_list(uint32_t flags)
      | ^~~~~~~~~~~~~~~
cc1: some warnings being treated as errors

```

</details>

Reply:
Hmm, so looking at the log and the resulting binary, I successful built the _decimal module with cpython 3.8.6 on an ancient linux distro with gcc 9 against a static copy of mpdecimal 2.5.0. I didn't try cpython 3.7.z or a shared copy of mpdecimal. I can give these a try if you want.

Does the decimal module work without mpdecimal? (presumably falling back onto a pure python implementation) If so, then I think that mpdecimal should be made an optional dependency, enabled by default.

Reply:
> Hmm, so looking at the log and the resulting binary, I successful built the _decimal module with cpython 3.8.6 on an ancient linux distro with gcc 9 against a static copy of mpdecimal 2.5.0. I didn't try cpython 3.7.z or a shared copy of mpdecimal. I can give these a try if you want.

I just tried 3.8.6 and that one can indeed build with mpdecimal 2.5.0 successfully.
So it's just versions <=3.7 that have difficulty.

> Does the decimal module work without mpdecimal? (presumably falling back onto a pure python implementation) If so, then I think that mpdecimal should be made an optional dependency, enabled by default.

Yes, see https://github.com/python/cpython/blob/master/Lib/decimal.py
There is a pure python fallback.

Reply:
The error message should be fixed to indicate that this applies to cpython versions < 3.8.0.